Sources/NetworkProtection/Settings/TunnelSettings.swift

```diff
@@ -275,6 +330,54 @@ public final class TunnelSettings {
         }
     }
 
+    // MARK: - Show in Menu Bar
+
+    public var showInMenuBarPublisher: AnyPublisher<Bool, Never> {
+        defaults.networkProtectionSettingShowInMenuBarPublisher
+    }
+
+    public var showInMenuBar: Bool {
+        get {
+            defaults.networkProtectionSettingShowInMenuBar
+        }
+
+        set {
+            defaults.networkProtectionSettingShowInMenuBar = newValue
+        }
+    }
+
+    // MARK: - Show VPN Settings
+
+    public var showVPNSettingsPublisher: AnyPublisher<Bool, Never> {
+        defaults.networkProtectionSettingShowVPNSettingsPublisher
+    }
+
+    public var showVPNSettings: Bool {
+        get {
+            defaults.networkProtectionSettingShowVPNSettings
+        }
+
+        set {
+            defaults.networkProtectionSettingShowVPNSettings = newValue
+        }
+    }
+
+    // MARK: - Notify Status Changes
+
+    public var notifyStatusChangesPublisher: AnyPublisher<Bool, Never> {
+        defaults.networkProtectionNotifyStatusChangesPublisher
+    }
+
+    public var notifyStatusChanges: Bool {
+        get {
+            defaults.networkProtectionNotifyStatusChanges
+        }
+
+        set {
+            defaults.networkProtectionNotifyStatusChanges = newValue
+        }
+    }
```

Comment:
I wonder if a Macro could be used to save space with this boilerplate code. (I wouldn't actually investigate that, since we can't use Xcode 15 everywhere yet - just thinking out loud.)

Reply:
Yes, I have considered that.  I also considered using a protocol, but there are going to be some unique challenges.

The way I currently see `TunnelSettings` is that it's showing us through it's organic growth that it may need a refactoring to remove some of the code duplication.